lib/inferno/apps/cli/execute.rb

```diff
@@ -0,0 +1,287 @@
+      end
+
+      def create_params(test_session, runnable)
+        {
+          test_session_id: test_session.id,
+          runnable_id_key => runnable.id,
+          inputs: thor_hash_to_inputs_array(options[:inputs])
+        }
+      end
+
+      def serialize(entity)
+        case entity.class.to_s
+        when 'Array'
+          JSON.pretty_generate(entity.map { |item| JSON.parse serialize(item) })
+        when lambda { |x|
+               defined?(x.constantize) && defined?("Inferno::Web::Serializers::#{x.split('::').last}".constantize)
+             }
+          "Inferno::Web::Serializers::#{entity.class.to_s.split('::').last}".constantize.render(entity)
+        else
+          raise StandardError, "CLI does not know how to serialize #{entity.class}"
+        end
+      end
+
+      def verbose_print(*args)
+        print(COLOR.dim(*args)) if options[:verbose]
+      end
+
+      def verbose_puts(*args)
+        args.push("\n")
+        verbose_print(*args)
+      end
+
+      def format_id(result)
+        result.runnable.id
+      end
+
+      def format_messages(result)
+        result.messages.map do |message|
+          "\n\t\t#{message.type}: #{message.message}"
+        end.join
+      end
+
+      def format_requests(result)
+        result.requests.map do |req_res|
+          "\n\t\t#{req_res.status} #{req_res.verb.upcase} #{req_res.url}"
+        end.join
+      end
+
+      def format_session_data(result, attr)
+        json = result.send(attr)
+        return '' if json.nil?
+
+        JSON.parse(json).map do |hash|
+          "\n\t\t#{hash['name']}: #{hash['value']}"
+        end.join
+      end
+
+      def format_inputs(result)
+        format_session_data(result, :input_json)
+      end
+
+      def format_outputs(result)
+        format_session_data(result, :output_json)
+      end
+
+      def format_result(result) # rubocop:disable Metrics/CyclomaticComplexity
+        case result.result
+        when 'pass'
+          COLOR.bold.green(CHECKMARK, ' pass')
+        when 'fail'
+          COLOR.bold.red 'X fail'
+        when 'skip'
+          COLOR.yellow '* skip'
+        when 'omit'
+          COLOR.blue '* omit'
+        when 'error'
+          COLOR.magenta 'X error'
+        when 'wait'
+          COLOR.bold '. wait'
+        when 'cancel'
+          COLOR.red 'X cancel'
+        when 'running'
+          COLOR.bold '- running'
+        else
+          raise StandardError.new, "Unrecognized result #{result.result}"
+        end
+      end
+
+      def verbose_print_json_results(results)
+        verbose_puts BAR
+        verbose_puts 'JSON Test Results:'
+        verbose_puts BAR
+        verbose_puts serialize(results)
+        verbose_puts BAR
+      end
+
+      def print_color_results(results)
+        puts BAR
+        puts 'Colored Test Results:'
+        puts BAR
+        results.each do |result|
+          print format_id(result), ': ', format_result(result), "\n"
```

Comment:
For this let's do short id + title.

Reply:
I basically did short_id + (short_title || title || id) if that's okay